crates/transaction-pool/src/pool/parked.rs

```diff
@@ -195,7 +196,10 @@ impl<T: ParkedOrd> ParkedPool<T> {
         {
             // NOTE: This will not panic due to `!last_sender_transaction.is_empty()`
             let sender_id = self.last_sender_submission.last().expect("not empty").sender_id;
-            let list = self.get_txs_by_sender(sender_id);
+            // Utilize SmallVec to efficiently handle up to 16 transactions
+            // per sender, adhering to the TXPOOL_MAX_ACCOUNT_SLOTS_PER_SENDER limit.
+            let list: SmallVec<[TransactionId; TXPOOL_MAX_ACCOUNT_SLOTS_PER_SENDER]> =
+                self.get_txs_by_sender(sender_id).into();
```

Comment:
Actually here it doesn't make any sense, `get_txs_by_sender` already returns a Vec and we just iterate over it immediatly after. Should this be changed in the `get_txs_by_sender` return type? @mattsse

Reply:
yeah this should be the return type of the function